### PR TITLE
Add SimpleEventDispatcher::getClassesEvents()

### PR DESCRIPTION
**Pull request by vjik**
| Q             | A
| ------------- | ---
| Is bugfix?    | ❌
| New feature?  | ✔️
| Breaks BC?    | ❌
| Fixed issues  | -

### CODE REVIEW

File: src/EventDispatcher/SimpleEventDispatcher.php [hunk 1]
```diff
@@ -43,6 +43,14 @@ public function getEvents(): array
         return $this->events;
     }
 
+    public function getClassesEvents(): array
```
Comment (roxblnfk) [inline src/EventDispatcher/SimpleEventDispatcher.php]:
```suggestion
    /**
     * @psalm-return array<int, class-string>
     */
    public function getClassesEvents(): array
```

**Review by roxblnfk — APPROVED** [review]
It can be useful 👍

File: src/EventDispatcher/SimpleEventDispatcher.php [hunk 2]
```diff
@@ -43,6 +43,14 @@ public function getEvents(): array
...
+    {
+        return array_map(
+            static fn ($event) => get_class($event),
+            $this->events
+        );
```
Comment (roxblnfk) [inline src/EventDispatcher/SimpleEventDispatcher.php]:
```suggestion
        return array_map('get_class', $this->events);
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -8,6 +8,7 @@
 ## 1.1.0 February 22, 2021
 
 - Enh #23: Add `SimpleEventDispatcher::clearEvents()` that clear all events in event dispatcher (vjik)
+- Enh #24: Add `SimpleEventDispatcher::getClassesEvents()` that return classes of dispatched events (vjik)
```
Comment (samdark) [inline CHANGELOG.md]: 1.1.0 was already tagged.

File: src/EventDispatcher/SimpleEventDispatcher.php (hunk 1 above)
Comment (samdark) [inline src/EventDispatcher/SimpleEventDispatcher.php]:
```suggestion
    public function getEventClasses(): array
```

File: tests/EventDispatcher/SimpleEventDispatcherTest.php [hunk 4]
```diff
@@ -225,6 +226,21 @@ public function testGetEvents(): void
         $this->assertSame([$event1, $event2, $event3], $dispatcher->getEvents());
     }
 
+    public function testGetClassesEvents(): void
```
Comment (samdark) [inline tests/EventDispatcher/SimpleEventDispatcherTest.php]:
```suggestion
    public function testGetEventClasses(): void
```

**Review by samdark — CHANGES_REQUESTED** [review]

File: src/EventDispatcher/SimpleEventDispatcher.php (hunk 1 above)
Comment (vjik) [reply src/EventDispatcher/SimpleEventDispatcher.php]: done

File: src/EventDispatcher/SimpleEventDispatcher.php (hunk 2 above)
Comment (vjik) [reply src/EventDispatcher/SimpleEventDispatcher.php]: done

File: CHANGELOG.md (hunk 3 above)
Comment (vjik) [reply CHANGELOG.md]: done

File: src/EventDispatcher/SimpleEventDispatcher.php (hunk 1 above)
Comment (vjik) [reply src/EventDispatcher/SimpleEventDispatcher.php]: done

File: tests/EventDispatcher/SimpleEventDispatcherTest.php (hunk 4 above)
Comment (vjik) [reply tests/EventDispatcher/SimpleEventDispatcherTest.php]: done

**Review by samdark — APPROVED** [review]

**Pull request merged**